### ISSUE TITLE
maxStorageBufferRange is 0 on M1 chip

### ISSUE THREAD

**Issue by jhvst** [KhronosGroup/MoltenVK#1245]
Hi there,

I am trying to run GPGPU workload on M1 Mac, but I am running into a problem where the validation layer reports the following thing:

```
"Validation Error: [ VUID-VkWriteDescriptorSet-descriptorType-00333 ] Object 0: handle = 0x410000000041, type = VK_OBJECT_TYPE_DESCRIPTOR_SET; | MessageID = 0xf2fc081c | vkUpdateDescriptorSets() failed write update validation for VkDescriptorSet 0x410000000041[] with error: Write update to VkDescriptorSet 0x410000000041[] allocated with VkDescriptorSetLayout 0x3a000000003a[] binding #0 failed with error message: Attempted write update to buffer descriptor failed due to: For buffer VkBuffer 0xf700000000f7[] VkDescriptorBufferInfo range is VK_WHOLE_SIZE but effective range (4584096) is greater than this device\'s maxStorageBufferRange (0). The Vulkan spec states: If descriptorType is VK_DESCRIPTOR_TYPE_STORAGE_BUFFER or VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC, the range member of each element of pBufferInfo, or the effective range if range is VK_WHOLE_SIZE, must be less than or equal to VkPhysicalDeviceLimits::maxStorageBufferRange (https://vulkan.lunarg.com/doc/view/1.2.162.1/mac/1.2-extensions/vkspec.html#VUID-VkWriteDescriptorSet-descriptorType-00333)"
```

The same code works fine on desktop platforms. 

`vkconfig` reports the following:

```
Vulkan Development Status:
- VULKAN_SDK environment variable not set
- Vulkan Loader version: 1.2.162
- Custom Layers Paths: None
- Available Layers:
    - VK_LAYER_LUNARG_override (1.2.162)
    - VK_LAYER_LUNARG_api_dump (1.2.162)
    - VK_LAYER_KHRONOS_validation (1.2.162)
    - VK_LAYER_LUNARG_device_simulation (1.2.162)
- Physical Devices:
    - Apple M1 (Discrete GPU) with Vulkan 1.1.162
```    

Any idea what might be the cause of the issue? Anything I could do the further help out on this?

**Comment by billhollings** [comment]
This was recently reported in issue #1240, and was fixed in PR #1242.

Can you pull in the latest and try again?

**Comment by jhvst** [comment]
Great to hear there is a fix already! I only searched the issues with my maxStorageBufferRange so I ended up missing that PR. I assume the PR will fix my problems, so I will close the issue and re-open if it wont fix my situation.

Thanks!
